### PR TITLE
Migrate array tutorial to pure python mode

### LINKED ISSUE #4187: [ENH] Pythonise the documentation!
The current documentation stems mostly from a time when Pure Python Mode wasn't even being thought of and aims to present the special `cdef` syntax that Cython uses. The Pure Python Syntax is only presented on the [pure Python tutorial page](http://docs.cython.org/en/latest/src/tutorial/pure.html).

Today, more and more new users come from a Python background that uses type annotations. We should accommodate them by presenting both syntax variants hand in hand throughout the documentation.

If you want to help out, please do the following:

1) Grab a documentation page and duplicate the code examples, putting the Pure Python variant *before* the `cdef` variant.
This currently excludes the pages that refer to the integration of external C or C++ code, since some of the required language features are not available in Pure Python code/mode.

Example:
```cython
cdef int add1(int x) except -1:
    return x + 1
```
should become
```cython
@cython.exceptval(-1)
def add1(x: cython.int):
    return x + 1

cdef int add1(int x) except -1:
    return x + 1
```

2) Read the surrounding text to see if there is anything that refers to the specific syntax constructs. Make sure everything is well explained to the readers so that they can easily understand the two examples.

3) Put the (proposed) following paragraph near the top of the page to show a reference to `import cython` and briefly explain the difference between the two syntax variants:

```
This page uses two different syntax variants: the Cython specific `cdef` syntax
and static Cython type declarations in pure Python code. To make good use
of the latter, you need the special `cython` module, which you can import with

.. code-block: python
      import cython

in the Python module that you want to compile.
```

Don't forget to link back to this issue in any PRs that you submit.

### PR DESCRIPTION
**Pull request by matusvalo**
Part of #4187

### CODE REVIEW

File: docs/src/tutorial/array.rst [hunk 1]
```diff
@@ -55,13 +79,29 @@ it is possible to create a new array with the same type as a template,
...
+    .. group-tab:: Pure Python
+
+        .. literalinclude:: ../../examples/tutorial/array/resize.py
+
+    .. group-tab:: Cython
+
+        .. literalinclude:: ../../examples/tutorial/array/resize.pyx
 
 
 API reference
```
Comment (matusvalo) [inline docs/src/tutorial/array.rst:107]: I am not sure whether we want to have migrated function definition in API reference or we will stick just with Cython syntax...

Comment (scoder) [reply docs/src/tutorial/array.rst:110]: Good question. Not sure. I mean, we shouldn't _require_ users to learn both syntax variants. So I think presenting both, as for the examples above, seems like a good idea.

**Comment by scoder** [comment]
Looks good to me.

**Comment by scoder** [comment]
Just noticed that this was already converted in #4431. Sorry for the duplication. @0dminnimda had the same question, BTW.

**Comment by matusvalo** [comment]
> Just noticed that this was already converted in #4431. Sorry for the duplication. @0dminnimda had the same question, BTW.

Oh nuts! I missed that!

**Comment by 0dminnimda** [comment]
> Oh nuts! I missed that!

yes it is rather unfortunate

**Comment by scoder** [comment]
Superseded by #4431

**Pull request closed without merging**